discord/ext/commands/core.py

```diff
@@ -960,47 +962,95 @@ def signature(self):
 
         result = []
         for name, param in params.items():
-            greedy = isinstance(param.annotation, Greedy)
-            optional = False  # postpone evaluation of if it's an optional argument
-
-            # for typing.Literal[...], typing.Optional[typing.Literal[...]], and Greedy[typing.Literal[...]], the
-            # parameter signature is a literal list of it's values
-            annotation = param.annotation.converter if greedy else param.annotation
-            origin = getattr(annotation, '__origin__', None)
-            if not greedy and origin is Union:
-                none_cls = type(None)
-                union_args = annotation.__args__
-                optional = union_args[-1] is none_cls
-                if len(union_args) == 2 and optional:
-                    annotation = union_args[0]
-                    origin = getattr(annotation, '__origin__', None)
-
-            if origin is Literal:
-                name = '|'.join(f'"{v}"' if isinstance(v, str) else str(v) for v in annotation.__args__)
-            if param.default is not param.empty:
-                # We don't want None or '' to trigger the [name=value] case and instead it should
-                # do [name] since [name=None] or [name=] are not exactly useful for the user.
-                should_print = param.default if isinstance(param.default, str) else param.default is not None
-                if should_print:
-                    result.append(f'[{name}={param.default}]' if not greedy else
-                                  f'[{name}={param.default}]...')
-                    continue
+            flag = isinstance(param.annotation, type) and issubclass(param.annotation, FlagConverter)
```

Comment:
For performance reasons this check is done using `hasattr(x, '__comands_is_flag__')`.